.github/workflows/release-prep.yml

```diff
@@ -0,0 +1,85 @@
+    inputs:
+      release_version:
+        description: "The version to release"
+        required: true
+        default: "0.0.0"
+      base_branch:
+        description: "The base branch to create the release branch from"
+        required: true
+        default: "main"
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  release:
+    # environment: release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7 # v2.10.1
+        with:
+          egress-policy: audit
+
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.base_branch }}
+      - name: Bump version.py and create PR
+        env:
+          PR_BRANCH: pre-release/${{ inputs.release_version }}
+          VERSION: ${{ inputs.release_version }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # Create branch
+          git checkout -b pre-release/${VERSION}
+
+          # Split version number into major, minor, and micro
+          MAJOR=$(echo $VERSION | cut -d. -f1)
+          MINOR=$(echo $VERSION | cut -d. -f2)
+          MICRO=$(echo $VERSION | cut -d. -f3)
+
+          # Write on to the version.py file
+          sed -i "s/major: int = \([0-9]\+\)/major: int = $MAJOR/g" heat/core/version.py
+          sed -i "s/minor: int = \([0-9]\+\)/minor: int = $MINOR/g" heat/core/version.py
+          sed -i "s/micro: int = \([0-9]\+\)/micro: int = $MICRO/g" heat/core/version.py
+          sed -i "s/extension: str = .*/extension: str = None/g" heat/core/version.py
+
+          # Git configuration with anonymous user and email
+          git config --global user.email ""
+          git config --global user.name "Heat Release Bot"
+
+          # Commit the changes
+          git add heat/core/version.py
+          git commit -m "Bump version to $VERSION"
+
+          # Push the changes
+          git push --set-upstream origin pre-release/${VERSION}
+
+          # Create PR for release
+          gh pr create \
+          --base release \
+          --head ${{ env.PR_BRANCH }} \
+          --title "Heat ${{ env.VERSION }} - Release" \
+          --body "Pre-release branch for Heat ${{ env.VERSION }}.
+          Any release work should be done on this branch, and then merged into the release branch and main, following git-flow.
+
+          TODO:
+          - [x] Update version.py
+          - [ ] update the Requirements section on README.md if needed
+          - [ ] Update CITATION.cff if needed
+          - [ ] Ensure the Changelog is up to date
+
+          [1]: https://github.com/peter-evans/create-pull-request" \
+          --label invalid
+
+          # Create PR for main
+          gh pr create --base main \
+          --head ${{ env.PR_BRANCH }} \
+          --title "Heat ${{ env.VERSION }} - Main" \
+          --body "Copy of latest pre-release PR targeting release." \
+          --label invalid
```

Comment:
Should this PR be created before release or within the post-release workflow?

Reply:
I thought it could be useful to have it from the start, as it might highlight merge conflicts and CI errors at the same time in both target branches at the same time, so we would find the earlier. 

We could ignore until after the release PR is merged, or move this to a post-release workflow

Reply:
Perfect, let's try it out